### PR TITLE
add leadid fields

### LINKED ISSUE #59: Please add all LeadiD Fields
Hi Jonathan, can you ensure that we can send all LeadiD fields to Velocify? Attached is a spreadsheet that shows Velocify's field names. Please let me know if you see any gaps or have questions. We'd like to get this in place quickly so Bridgeview can roll out the integration.
[Leadid_AP_Velocify_fields.txt.xlsx](https://github.com/activeprospect/leadconduit-integration-velocify/files/195685/Leadid_AP_Velocify_fields.txt.xlsx)

### PR DESCRIPTION
**Pull request by jgrayson676**
fixes #59

### CODE REVIEW

File: src/mortgage.coffee [hunk 1]
```diff
@@ -272,6 +379,10 @@ request = (vars) ->
     # null: 'lead.relationship'
     # null: 'lead.relationship_description'
     # gift_funds: lead.gift_funds
+  content['consumer_intent_score.outcome'] = vars.consumer_intent_score?.outcome
+  content['consumer_intent_score.id'] = vars.consumer_intent_score?.id
+  content['consumer_intent_score.score'] = vars.consumer_intent_score?.score
+  content['consumer_intent_score.reason'] = vars.consumer_intent_score?.reason
```
Comment (cgrayson) [inline src/mortgage.coffee]: Why do these 4 separately, instead of the `key: value` way above?

File: spec/mortgage_spec.coffee [hunk 2]
```diff
@@ -176,7 +176,28 @@ describe 'Outbound mortgage request', ->
...
+    it 'should map LeadiD fields', ->
+      lead =
+        source:
+          name: 'Leads R Us'
+        lead:
+          fields.buildLeadVars(leadid_id: '213')
+        authentic: 'true'
+        consumer_five_minutes: '3'
+        prominence_rule: 'green'
+        consumer_intent_score:
```
Comment (cgrayson) [inline spec/mortgage_spec.coffee]: Maybe I'm misunderstanding the purpose, but wouldn't all the LeadiD data have to have been appended, and so under the `leadid.` append key?

File: src/mortgage.coffee [hunk 3]
```diff
@@ -376,6 +487,114 @@ request.variables = ->
     { name: 'lead.age', type: 'number', required: false, description: '' }
     { name: 'source.name', type: 'string', required: false, description: 'Name of the source that provided the lead' }
     { name: 'lead.company.name', type:'string', required: false, description:'Name of company/employer'}
+    #LeadiD fields
```
Comment (cgrayson) [inline src/mortgage.coffee]: Depending on the answer about appended-data only, this block may not be needed. (I could even see an argument that we shouldn't _let_ users map values for these supposedly-from-LeadiD values.)

File: src/mortgage.coffee (hunk 1 above)
Comment (jgrayson676) [inline src/mortgage.coffee]: `querystring.encode` doesn't work with nested elements, so they were being encoded as `consumer_intent_score=` when declared like the rest of the fields

File: spec/mortgage_spec.coffee (hunk 2 above)
Comment (jgrayson676) [inline spec/mortgage_spec.coffee]: I think you're right that that's a better solution. Though it looks like most of the appended data from the LeadiD integration [isn't being prefixed with `leadid.`](https://github.com/activeprospect/leadconduit-integration-leadid/blob/master/src/audit.coffee#L140) unless that's something that happens automatically.

Comment (cgrayson) [inline spec/mortgage_spec.coffee]: Yeah, that does happen automatically, based on module name.

File: src/mortgage.coffee [hunk 4]
```diff
@@ -103,6 +103,113 @@ request = (vars) ->
     property_purchase_year: lead.mortgage?.property?.purchased_year
 
     #
+    # LeadID fields
+    #
+
+    authentic: vars.leadid?.authentic
```
Comment (cgrayson) [inline src/mortgage.coffee]: I wonder if these could be set by iterating over all the attributes of `vars.leadid` (with an exception or two, like `token`)?

**Comment by cgrayson** [comment]
Otherwise :+1:

**Pull request merged**